Microsoft.DotNet.Interactive/PackageRestoreContext.cs

```diff
@@ -13,58 +13,71 @@
 using Microsoft.DotNet.Interactive.Utility;
 using Pocket;
 using static Pocket.Logger;
+using Interactive.DependencyManager;
+using System.Runtime.InteropServices.ComTypes;
+using System.Collections.ObjectModel;
+using System.Globalization;
 
 namespace Microsoft.DotNet.Interactive
 {
     public class PackageRestoreContext : IDisposable
     {
+        private const string restoreTfm = "netcoreapp3.1";
+        private const string packageKey = "nuget";
+        private readonly string _scriptExtension;
         private readonly ConcurrentDictionary<string, PackageReference> _requestedPackageReferences = new ConcurrentDictionary<string, PackageReference>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, ResolvedPackageReference> _resolvedPackageReferences = new Dictionary<string, ResolvedPackageReference>(StringComparer.OrdinalIgnoreCase);
         private readonly HashSet<string> _restoreSources = new HashSet<string>();
-        private readonly Lazy<DirectoryInfo> _lazyDirectory;
+        private readonly Lazy<DependencyProvider> _lazyDependencies;
+        private readonly Lazy<IDependencyManagerProvider> _lazyIdm;
 
-        public PackageRestoreContext()
+        public PackageRestoreContext(string scriptExtension)
         {
-            _lazyDirectory = new Lazy<DirectoryInfo>(() =>
+            _scriptExtension = scriptExtension;
+            _lazyDependencies = new Lazy<DependencyProvider>(new DependencyProvider(AssemblyProbingPaths, NativeProbingRoots));
+            _lazyIdm = new Lazy<IDependencyManagerProvider>(_lazyDependencies.Value.TryFindDependencyManagerByKey(Enumerable.Empty<string>(), "", ReportError, packageKey));
+        }
+
+        private IEnumerable<string> AssemblyProbingPaths()
+        {
+            foreach (var package in _resolvedPackageReferences.Values)
             {
-                var dir = new DirectoryInfo(
-                    Path.Combine(
-                        Paths.UserProfile,
-                        ".net-interactive-csharp",
-                        Guid.NewGuid().ToString("N")));
+                foreach (var fi in package.AssemblyPaths)
+                    yield return fi.FullName;
+            }
+        }
 
-                if (!dir.Exists)
+        private IEnumerable<string> NativeProbingRoots ()
+        {
+            foreach (var package in _resolvedPackageReferences.Values)
+            {
+                foreach (var di in package.ProbingPaths)
                 {
-                    dir.Create();
+                    yield return di.FullName;
                 }
-
-                return dir;
-            });
-
-            AssemblyLoadContext.Default.Resolving += OnResolving;
+            }
         }
 
-        private Assembly OnResolving(AssemblyLoadContext loadContext, AssemblyName assemblyName)
+        private ResolvingErrorReport ReportError = (ErrorReportType errorType, int code, string message) =>
```

Comment:
Any reason for this to be a field rather than a method?

Reply:
It's a callback the ReportErrorFunc is invoked from within TryFindDependencyManagerByKey.  So the func is passed in as an argument to TryFindDependencyManagerByKey.